datadog/resource_datadog_dashboard.go

```diff
@@ -5086,6 +5093,111 @@ func buildTerraformProcessQuery(datadogQuery datadogV1.ProcessQueryDefinition) m
 	return terraformQuery
 }
 
+// APM Resources Query
+func getApmStatsQuerySchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"service": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"name": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"env": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"primary_tag": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"row_type": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"resource": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+				"columns": {
+					Type:     schema.TypeList,
+					Optional: true,
+					Elem:     &schema.Schema{Type: schema.TypeString},
+				},
+			},
+		},
+	}
+}
+
+func buildDatadogApmStatsQuery(terraformQuery map[string]interface{}) *datadogV1.ApmStatsQueryDefinition {
+	datadogQuery := datadogV1.NewApmStatsQueryDefinitionWithDefaults()
+	if v, ok := terraformQuery["service"].(string); ok && len(v) != 0 {
+		datadogQuery.SetService(v)
+	}
+	if v, ok := terraformQuery["name"].(string); ok && len(v) != 0 {
+		datadogQuery.SetName(v)
+	}
+	if v, ok := terraformQuery["env"].(string); ok && len(v) != 0 {
+		datadogQuery.SetEnv(v)
+	}
+	if v, ok := terraformQuery["primary_tag"].(string); ok && len(v) != 0 {
+		datadogQuery.SetPrimaryTag(v)
+	}
+	if v, ok := terraformQuery["row_type"].(datadogV1.ApmStatsQueryRowType); ok && len(v) != 0 {
```

Comment:
I was getting an error for `terraformQuery["row_type"].(string)` and I think this is the solution, but please let me know if there's a more correct way

Reply:
I think it might more make sense to do `datadogQuery.SetRowType(datadogV1.ApmStatsQueryRowType(v))`.